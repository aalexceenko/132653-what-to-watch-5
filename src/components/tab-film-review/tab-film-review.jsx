import React from "react";
import PropTypes from "prop-types";
// import {review} from "../../mocks/films";
// import {getDuration} from "../../utils";

import {reviewType} from '../../types/review';
import {getDateRevieFormat, sortReviewRating} from "../../utils";

const createReviewTemplate = (reviews) => {
  return (
    <React.Fragment>
      {Object.values(reviews.map((review, i) => {
        const {message, author, date, rating} = review;
        return (
          <div className="review" key={i}>
            <blockquote className="review__quote">
              <p className="review__text">{message}</p>
              <footer className="review__details">
                <cite className="review__author">{author}</cite>
                <time className="review__date" dateTime="2016-12-24">{getDateRevieFormat(date)}</time>
              </footer>
            </blockquote>
            <div className="review__rating">{rating}</div>
          </div>
        );
      }))}
    </React.Fragment>
  );
};

const TabFilmReview = ({reviews}) => {
  console.log(reviews);
  let sortReviews = reviews.sort(sortReviewRating);
  console.log(sortReviews);

  console.log(Object.keys(reviews).length);

  let rightColumn;
  let leftColumn;

  let lengthReviews = Object.keys(reviews).length;
  let midleCount = Math.round(lengthReviews / 2);

  leftColumn = reviews.slice(0, midleCount);
  rightColumn = reviews.slice(midleCount);

  return (
    <div className="movie-card__reviews movie-card__row">
      <div className="movie-card__reviews-col">

        {createReviewTemplate(leftColumn)}

      </div>
      <div className="movie-card__reviews-col">

        {createReviewTemplate(rightColumn)}

      </div>
    </div>
  );
};

TabFilmReview.propTypes = {
  // runtime: PropTypes.number.isRequired,
  // year: PropTypes.number.isRequired,
  // genre: PropTypes.string.isRequired,
  // director: PropTypes.string.isRequired,
  // actors: PropTypes.array.isRequired,
  // review: reviewType.isRequired,
  reviews: PropTypes.arrayOf(reviewType).isRequired,

};

export default TabFilmReview;
