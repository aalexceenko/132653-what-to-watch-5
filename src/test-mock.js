import {getFilmRank} from "./utils";

export const VISIBLE_FILMS_COUNT = 8;
export const ID = 1;
export const VIDEO_CURRENT_TIME = 0;
export const PROGRESS_POSITION = 0;
export const AUTHORIZATION_STATUS = `AUTH`;

export const IS_VIDEO = true;
export const IS_PLAYING = true;
export const IS_IN_MY_LIST = false;
export const TAB = `details`;
export const PREVIEW = `img/aviator.jpg`;
export const PREVIEW_VIDEO = `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
export const TITLE = `Bohemian Rhapsody`;
export const ACTIVE_GENRE = `Comedy`;
export const GENRES = [`Comedy`, `Crime`, `Documentary`, `Drama`, `Horror`, `Kids & Family`];
export const MATCH = {
  params: {
    id: 1
  }
};

export const FILMS = [
  {
    description: `A young man who was sentenced to seven years`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 1,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    year: 2008,
    ratingText: getFilmRank(3.6),
    myList: false,
    comments: false,
  },
  {
    description: `A young man who was`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 2,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    year: 2008,
    ratingText: getFilmRank(3.6),
    myList: false,
    comments: false,
  },
];

export const FILM = {
  description: `A young man who was sentenced to seven years`,
  rating: 3.6,
  director: `Nicolas Winding Refn`,
  genre: `Action`,
  id: 3,
  title: `Bronson`,
  actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  votes: 109661,
  runtime: 92,
  year: 2008,
  ratingText: getFilmRank(3.6),
  myList: false,
  comments: false,
};

export const REVIEWS = [
  {
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    user: {
      name: `Christina`,
      id: 1,
    },
    comment: `This film is an experience and i has already seen it 4 times`,
  },
  {
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    user: {
      name: `Christina`,
      id: 2,
    },
    comment: `This film is an experience and i has already seen it 4 times`,
  }
];
