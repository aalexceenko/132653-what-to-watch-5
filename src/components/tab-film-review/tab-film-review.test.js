import React from "react";
import renderer from "react-test-renderer";
import {TabFilmReview} from "./tab-film-review";
import {ID, REVIEWS} from "../../test-mock";


it(`Should TabFilmReview render correctly`, () => {
  const tree = renderer
    .create(
        <TabFilmReview
          reviews={REVIEWS}
          getReviews={() => {}}
          id={ID}
        />

    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
