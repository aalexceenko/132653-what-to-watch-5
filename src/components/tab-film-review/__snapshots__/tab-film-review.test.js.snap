// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should TabFilmReview render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        />
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            October 9, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.3
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        />
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            October 9, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.3
      </div>
    </div>
  </div>
</div>
`;
